File: lumina-storage-backend/src/schemas/system.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class SystemConfigCreateRequest(BaseModel):
    key: str
    value: Any
    description: str | None = None
    is_public: bool = False


class SystemConfigUpdateRequest(BaseModel):
    value: Any | None = None
    description: str | None = None
    is_public: bool | None = None


class SystemConfigResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    key: str
    value: Any
    description: str | None
    is_public: bool
    updated_at: datetime
    updated_by_id: uuid.UUID | None


class PublicConfigResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    key: str
    value: Any

File: lumina-storage-frontend/src/app/types/system.ts
export interface SystemConfigResponse {
  id: number;
  key: string;
  value: unknown;
  description: string | null;
  is_public: boolean;
  updated_at: string;
  updated_by_id: string | null;
}

export interface PublicConfigResponse {
  key: string;
  value: unknown;
}

export interface SystemConfigCreateRequest {
  key: string;
  value: unknown;
  description?: string;
  is_public?: boolean;
}

export interface SystemConfigUpdateRequest {
  value?: unknown;
  description?: string;
  is_public?: boolean;
}
